api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1758,15 +1768,14 @@ type AzurePlatformSpec struct {
 	// +immutable
 	VnetID string `json:"vnetID,omitempty"`
```

Comment:
Here I would document the assumptions / requirements that are being made.

* The Subscription part specified in the VnetID has to match the value specified in the `SubscriptionID` attribute
* Does the resource group name part of the VnetID needs to match the `ResourceGroupName` value or the `NetworkResourceGroupName` value?
* Does the resource name part of the VnetID needs to match the value of the `VnetName` attribute?

Reply:
> The Subscription part specified in the VnetID has to match the value specified in the SubscriptionID attribute

I added this.

Reply:
> Does the resource group name part of the VnetID needs to match the ResourceGroupName value or the NetworkResourceGroupName value?

I added documentation to address this.

Reply:
> Does the resource name part of the VnetID needs to match the value of the VnetName attribute?

Yes, I added documentation to address this.

Reply:
Where has it been added? I currently don't see it :thinking:

Reply:
I see the following documentation in the AzurePlatformSpec type:
```
An existing cloud resource is expected to exist under the same SubscriptionID and
// within the same ResourceGroupName.
```

But is that really accurate? aren't we having some IDs that we allow to be in different resource group names? We should make the contract very clear